python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_asset_evaluations.py

```diff
@@ -65,23 +70,43 @@ class Meta:
         )
 
 
-def create_graphene_auto_materialize_condition(condition: AutoMaterializeCondition):
-    if isinstance(condition, FreshnessAutoMaterializeCondition):
-        return GrapheneFreshnessAutoMaterializeCondition(decisionType=condition.decision_type)
-    elif isinstance(condition, DownstreamFreshnessAutoMaterializeCondition):
-        return GrapheneDownstreamFreshnessAutoMaterializeCondition(
-            decisionType=condition.decision_type
-        )
-    elif isinstance(condition, ParentMaterializedAutoMaterializeCondition):
-        return GrapheneParentMaterializedAutoMaterializeCondition(
-            decisionType=condition.decision_type
+DAGSTER_NAMEDTUPLE_TO_GRAPHENE_MAP = {
+    FreshnessAutoMaterializeCondition: GrapheneFreshnessAutoMaterializeCondition,
+    DownstreamFreshnessAutoMaterializeCondition: GrapheneDownstreamFreshnessAutoMaterializeCondition,
+    ParentMaterializedAutoMaterializeCondition: GrapheneParentMaterializedAutoMaterializeCondition,
+    MissingAutoMaterializeCondition: GrapheneMissingAutoMaterializeCondition,
+    ParentOutdatedAutoMaterializeCondition: GrapheneParentOutdatedAutoMaterializeCondition,
+}
+
+
+def create_graphene_auto_materialize_condition(
+    condition: Union[AutoMaterializeCondition, Tuple[AutoMaterializeCondition, str]],
+    partitions_def: Optional[PartitionsDefinition],
+):
+    if partitions_def is not None:
+        if not len(condition) == 2 and isinstance(condition[1], str):
+            raise DagsterError(
+                f"Unexpected condition type {type(condition)} for partitions definition"
+                f" {partitions_def}"
+            )
+        auto_materialize_condition, serialized_subset = cast(
+            Tuple[AutoMaterializeCondition, str], condition
         )
-    elif isinstance(condition, MissingAutoMaterializeCondition):
-        return GrapheneMissingAutoMaterializeCondition(decisionType=condition.decision_type)
-    elif isinstance(condition, ParentOutdatedAutoMaterializeCondition):
-        return GrapheneParentOutdatedAutoMaterializeCondition(decisionType=condition.decision_type)
+        subset = partitions_def.deserialize_subset(serialized_subset)
+        partition_keys = subset.get_partition_keys()
     else:
-        check.failed(f"Unexpected condition type {type(condition)}")
+        auto_materialize_condition = cast(AutoMaterializeCondition, condition)
+        partition_keys = None
+
+    check.invariant(
+        type(auto_materialize_condition) in DAGSTER_NAMEDTUPLE_TO_GRAPHENE_MAP,
```

Comment:
sort of combining the above two comments, but I think we should get rid of this map